compiler/rustc_parse/messages.ftl

```diff
@@ -1,868 +0,0 @@
-    .suggestion = surround the type parameters with angle brackets
-
-parse_generics_in_path = unexpected generic arguments in path
-
-parse_help_set_edition_cargo = set `edition = "{$edition}"` in `Cargo.toml`
-parse_help_set_edition_standalone = pass `--edition {$edition}` to `rustc`
-parse_if_expression_missing_condition = missing condition for `if` expression
-    .condition_label = expected condition here
-    .block_label = if this block is the condition of the `if` expression, then it must be followed by another block
-
-parse_if_expression_missing_then_block = this `if` expression is missing a block after the condition
-    .add_then_block = add a block here
-    .condition_possibly_unfinished = this binary operation is possibly unfinished
-
-parse_in_in_typo =
-    expected iterable, found keyword `in`
-    .suggestion = remove the duplicated `in`
-
-parse_inappropriate_default = {$article} {$descr} cannot be `default`
-    .label = `default` because of this
-    .note = only associated `fn`, `const`, and `type` items can be `default`
-
-parse_inclusive_range_extra_equals = unexpected `=` after inclusive range
-    .suggestion_remove_eq = use `..=` instead
-    .note = inclusive ranges end with a single equals sign (`..=`)
-
-parse_inclusive_range_match_arrow = unexpected `>` after inclusive range
-    .label = this is parsed as an inclusive range `..=`
-    .suggestion = add a space between the pattern and `=>`
-
-parse_inclusive_range_no_end = inclusive range with no end
-    .suggestion_open_range = use `..` instead
-    .note = inclusive ranges must be bounded at the end (`..=b` or `a..=b`)
-
-parse_incorrect_parens_trait_bounds = incorrect parentheses around trait bounds
-parse_incorrect_parens_trait_bounds_sugg = fix the parentheses
-
-parse_incorrect_semicolon =
-    expected item, found `;`
-    .suggestion = remove this semicolon
-    .help = {$name} declarations are not followed by a semicolon
-
-parse_incorrect_use_of_await =
-    incorrect use of `await`
-    .parentheses_suggestion = `await` is not a method call, remove the parentheses
-    .postfix_suggestion = `await` is a postfix operation
-
-parse_incorrect_visibility_restriction = incorrect visibility restriction
-    .help = some possible visibility restrictions are:
-            `pub(crate)`: visible only on the current crate
-            `pub(super)`: visible only in the current module's parent
-            `pub(in path::to::module)`: visible only on the specified path
-    .suggestion = make this visible only to module `{$inner_str}` with `in`
-
-parse_inner_attr_explanation = inner attributes, like `#![no_std]`, annotate the item enclosing them, and are usually found at the beginning of source files
-parse_inner_attr_not_permitted = an inner attribute is not permitted in this context
-    .label_does_not_annotate_this = {parse_label_inner_attr_does_not_annotate_this}
```

Comment:
How do we support this scenario in the new format?
I think we may support it like this:

```rust
#[derive(Diagnostic)]
#[diag(label = "the label...", does_not_annotate_this = "the content can be refer...")]
pub(crate) struct ErrorStruct {
  ...

  pub sugg: Option<SubErrorStruct>,
}

#[derive(Subdiagnostic)]
pub(crate) struct SubErrorStruct {

    #[suggestion("{does_not_annotate_this}")
    pub span: Span
}
```